static/app/components/profiling/flamegraph.tsx

```diff
@@ -59,7 +81,149 @@ function Flamegraph(props: FlamegraphProps): ReactElement {
+        // left heavy has changed, we do want to persist the config view
+        if (previousView.flamegraph.leftHeavy === newView.flamegraph.leftHeavy) {
+          newView.setConfigView(
+            previousView.configView.withHeight(newView.configView.height)
+          );
+        }
+      }
+
+      return newView;
+    },
+    [flamegraph, flamegraphCanvas, flamegraphTheme]
+  );
+
+  useEffect(() => {
+    if (!flamegraphCanvas || !flamegraphView) {
+      return undefined;
+    }
+
+    const onConfigViewChange = (rect: Rect) => {
+      flamegraphView.setConfigView(rect);
+      canvasPoolManager.draw();
+    };
+
+    const onTransformConfigView = (mat: mat3) => {
+      flamegraphView.transformConfigView(mat);
+      canvasPoolManager.draw();
+    };
+
+    const onResetZoom = () => {
+      flamegraphView.resetConfigView(flamegraphCanvas);
+      canvasPoolManager.draw();
+    };
+
+    const onZoomIntoFrame = (frame: FlamegraphFrame) => {
+      flamegraphView.setConfigView(
+        new Rect(
+          frame.start,
+          frame.depth,
+          frame.end - frame.start,
+          flamegraphView.configView.height
+        )
+      );
+
+      canvasPoolManager.draw();
+    };
+
+    scheduler.on('setConfigView', onConfigViewChange);
+    scheduler.on('transformConfigView', onTransformConfigView);
+    scheduler.on('resetZoom', onResetZoom);
+    scheduler.on('zoomIntoFrame', onZoomIntoFrame);
+
+    return () => {
+      scheduler.off('setConfigView', onConfigViewChange);
+      scheduler.off('transformConfigView', onTransformConfigView);
+      scheduler.off('resetZoom', onResetZoom);
+      scheduler.off('zoomIntoFrame', onZoomIntoFrame);
+    };
+  }, [canvasPoolManager, flamegraphCanvas, flamegraphView, scheduler]);
+
+  useEffect(() => {
+    canvasPoolManager.registerScheduler(scheduler);
+    return () => canvasPoolManager.unregisterScheduler(scheduler);
+  }, [canvasPoolManager, scheduler]);
+
+  useEffect(() => {
+    if (
+      !flamegraphView ||
+      !flamegraphCanvas ||
+      !flamegraphMiniMapCanvas ||
+      !flamegraphCanvasRef ||
+      !flamegraphOverlayCanvasRef ||
+      !flamegraphMiniMapCanvasRef ||
+      !flamegraphMiniMapOverlayCanvasRef
+    ) {
+      return undefined;
+    }
+
+    const observer = watchForResize(
+      [
+        flamegraphCanvasRef,
+        flamegraphOverlayCanvasRef,
+        flamegraphMiniMapCanvasRef,
+        flamegraphMiniMapOverlayCanvasRef,
+      ],
+      () => {
+        const bounds = flamegraphCanvasRef.getBoundingClientRect();
+        setCanvasBounds(new Rect(bounds.x, bounds.y, bounds.width, bounds.height));
+
+        flamegraphCanvas.resizePhysicalSpace();
+        flamegraphMiniMapCanvas.resizePhysicalSpace();
+        flamegraphView.resizeConfigSpace(flamegraphCanvas);
+
+        canvasPoolManager.drawSync();
+      }
+    );
+
+    return () => observer.disconnect();
+  }, [
```

Comment:
Since the FlamegraphCanvas gets the canvas element, I think at some point we should think about splitting this down into two observers. Basically have
```
const bounds = flamegraphCanvasRef.getBoundingClientRect();
        setCanvasBounds(new Rect(bounds.x, bounds.y, bounds.width, bounds.height));

        flamegraphCanvas.resizePhysicalSpace();
        flamegraphMiniMapCanvas.resizePhysicalSpace();
        flamegraphView.resizeConfigSpace(flamegraphCanvas);
```
run inside the FlamegraphCanvas and only run `canvasPoolManager.drawSync();` here. The downside is that we will probably have some flickering as these two would not be guaranteed to run in order.